scripts/index.js

```diff
@@ -1,13 +1,94 @@
-import { generateDom, manipulation, renderVirtualDom } from './render/dom.js';
-// extract all the nodes from the body
-window.raqnVirtualDom = manipulation(generateDom(document.body.childNodes));
-// clear the body
-document.body.innerHTML = '';
-// append the nodes to the body after manipulation
-document.body.append(...renderVirtualDom(window.raqnVirtualDom));
-
-// EG callback to loadModules
-await Promise.all(window.initialization).then(() => {
-  // some after main modules loaded
-  console.log('All modules loaded');
-});
+import { generateVirtualDom, renderVirtualDom } from './render/dom.js';
+import { pageManipulation, templateManipulation } from './render/dom-manipulations.js';
+import { getMeta, metaTags, runTasks, isTemplatePage, previewModule } from './libs.js';
+
+await previewModule(import.meta);
+
+export default {
+  init() {
+    runTasks.call(
+      this, // all the tasks bellow will be bound to this object when called.
+      null,
+      this.generatePageVirtualDom,
+      this.pageVirtualDomManipulation,
+      this.loadAndProcessTemplate,
+      this.renderPage,
+      this.onPageRender,
+    );
+  },
+
+  generatePageVirtualDom() {
+    window.raqnVirtualDom = generateVirtualDom(document.body.childNodes);
+  },
+
+  async pageVirtualDomManipulation() {
+    await pageManipulation(window.raqnVirtualDom);
+  },
+
+  renderPage() {
+    const renderedDOM = renderVirtualDom(window.raqnVirtualDom);
+
+    if (renderedDOM) {
+      document.body.innerHTML = '';
+      document.body.append(...renderedDOM);
+    }
+  },
+
+  onPageRender() {
+    Promise.allSettled(window.initialization).then(() => {});
```

Comment:
what does this do?

Reply:
It would allow to run some code after all modules for the components on the page were loaded.
But not sure if we really needed this, it was there from Felipe's implementation and I kept it.

But now thinking about it I would change it from `window.initialization` to `window.raqnOnComponentsLoaded`

Reply:
will this really do anything without an `await`?

Reply:
I will delete it for now, we can brig it back when this will be really needed.